Add dotnet 6 runtime

yongliu-mdsol opened the pull request:
Aws Lambda supports dotnet 6 runtime.
Adding same support for localstack

yongliu-mdsol commented on .circleci/config.yml:
```diff
@@ -85,6 +85,7 @@ jobs:
             docker pull -q lambci/lambda:20191117-python3.6
             docker pull -q lambci/lambda:20191117-dotnetcore2.0
             docker pull -q lambci/lambda:dotnetcore3.1
+            docker pull -q lambci/lambda:dotnet6.0
```
waiting https://github.com/lambci/docker-lambda/pull/359 to get merged

dfangl replied:
Hi there! Lambci is pretty dead right now. Until we have the lambda rework in place, it might be a good solution to use the repository of mlupine: https://hub.docker.com/layers/docker-lambda/mlupin/docker-lambda/dotnet6/images/sha256-fc3edf2e50307a5c81e4da7690f6cb6427819279081e1d9b16e52a294bf21738?context=explore , as we do already with python3.9.

dfangl replied:
We would need to adjust the image though, in lambda_executor.py:1596.

yongliu-mdsol replied:
Finally got this working. 
@dfangl can you take a final review?

dfangl replied:
Please add the docker pull -q ... part back in for the mlupin image!

yongliu-mdsol replied:
👍🏻 added back

dfangl commented on tests/integration/awslambda/functions/dotnet6/src/dotnet6/Readme.md:
```diff
@@ -0,0 +1,49 @@
+# AWS Lambda Empty Function Project
```
It would be nice to rework this readme (remove all the autogenerated stuff), with just a short notice on how to build the commited zip archive (should be something like `dotnet lambda package`, right?)

yongliu-mdsol replied:
yup, I updated the readme.

dfangl commented on localstack/services/awslambda/lambda_executors.py:
```diff
@@ -1593,7 +1593,9 @@ def docker_image_for_lambda(cls, lambda_function: LambdaFunction):
             docker_image = "localstack/lambda-js"
-        if runtime == "python3.9" and docker_image == DEFAULT_LAMBDA_CONTAINER_REGISTRY:
+        elif (
+            runtime == "python3.9" or runtime == "dotnet6"
```
```suggestion
            runtime in ["python3.9", "dotnet6"]
```
could possibly also work. No real official style guidelines for this though, you might as well leave it.

yongliu-mdsol replied:
this actually reads better. I updated it

dfangl reviewed (changes requested):
All in all, thanks for taking this into your hands and creating this PR. Please just change the Readme accordingly, and add the `docker pull` command to the circleci stage again, for the mlupin image.
FYI: If it is alright with you, I would like to rebuild the zip deployment package myself, and commit it to your branch with a signed commit. I want to stress this is nothing against you personally, just for new contributors, I do not like to allow (potentially harmful) binary blobs to commit.

yongliu-mdsol commented:
@dfangl thanks again for the quick feedbacks. I've addressed your comments. 
Re recompling the zip file, I'm fine with that. Feel free to add any changes to make this PR better.

dfangl reviewed (approved):
LGTM now. Ready to merge once the build passes.
